x-pack/plugins/security_solution/public/detections/components/callouts/use_callout_storage.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { difference, fromPairs, identity } from 'lodash/fp';
+import { useCallback, useMemo } from 'react';
+import { useEffectOnce, useMap } from 'react-use';
+import { useMessagesStorage } from '../../../common/containers/local_storage/use_messages_storage';
+import { CallOutMessage } from './callout_types';
+
+export interface CallOutStorage {
+  getVisibleMessageIds: () => string[];
+  isVisible: (message: CallOutMessage) => boolean;
+  dismiss: (message: CallOutMessage) => void;
+}
+
+export const useCallOutStorage = (
```

Comment:
Is this a right place for this kind of code?
We have some `containers` folders in the plugin, and they contain hooks (mostly, I think).
On the other hand, this guy `useCallOutStorage` is tightly coupled to the components (dumb and smart) represented in this folder.

More broad question: how do we structure components and business logic they use?